lib/server/tcp-server.js

```diff
@@ -42,23 +43,24 @@ function Req(msg) {
 function Res(socket, ack) {
   this.ack = ack;
 
-  this.end = function() {
-    socket.write(VT + ack.toString() + FS + CR);
+  this.end = function(overrideAck) {
+    socket.write(VT + (overrideAck || ack).toString() + FS + CR);
```

Comment:
you can override the ack anywhere in the pipeline, as long as you do it before calling end.
so you don't need this.
```
app.use(function(req, res, next) {
   res.ack = something else.
})

```

Reply:
At the very least, doesn't the function need to be changed to call `this.ack.toString()` instead of `ack.toString()`?

But yes, if you prefer that rather than allowing a overrideAck arg, I'll make the change